webview/src/views/settings.svelte

```diff
@@ -50,22 +53,39 @@
 
 <section>
   {#await settingsLoadedPromise}
-    Loading...
+    <vscode-progress-ring />
+    <p>Loading...</p>
   {:then settings}
     <p>Configure client connection (for downloading histories)</p>
-    <form on:submit|preventDefault={saveSettings}>
-      <label for="address">Address</label>
-      <input type="text" required value={settings.address} name="address" />
-      <label for="tls">TLS?</label>
-      <input type="checkbox" name="tls" checked={settings.tls} />
-      <div />
-      <label for="clientCert">Client cert {settings.hasClientCert ? "(present)" : ""}</label>
-      <input type="file" name="clientCert" />
-      <div />
-      <label for="clientPrivateKey">Client private key {settings.hasClientPrivateKey ? "(present)" : ""}</label>
-      <input type="file" name="clientPrivateKey" />
-      <div />
-      <button type="submit">Submit</button>
+    <form id="settings-form">
+      <vscode-text-field type="text" required value={settings.address}>Address</vscode-text-field>
+      <div class="checkbox">
+        <vscode-checkbox checked={settings.tls}>TLS?</vscode-checkbox>
+      </div>
+
+      <label for="client-cert">Client cert {settings.hasClientCert ? "(present)" : ""}</label>
+      <FileInput id="client-cert" />
```

Comment:
Why do we need to add ids everywhere?

Reply:
For accessibility reasons, we want to explicitly associate the text from a  `<label/>` with an `<input/>` and therefore [the id on the input must match the for on the label](https://www.w3.org/WAI/tutorials/forms/labels/).

Reply:
You can tell by my comment that I was never a "real" frontend developer.